plugins/modules/mount.py

```diff
@@ -763,105 +764,116 @@ def main():
+                        res, msg = remount(module, args)
+                        changed = True
+                else:
+                    changed = True
+
+                    if not module.check_mode:
+                        res, msg = mount(module, args)
+
+                if res:
+                    # Not restoring fstab after a failed mount was reported as a bug,
+                    # ansible/ansible#59183
+                    # A non-working fstab entry may break the system at the reboot,
+                    # so undo all the changes if possible.
+                    try:
+                        write_fstab(module, backup_lines, args['fstab'])
+                    except Exception:
+                        pass
+
+                    try:
+                        for dirname in dirs_created[::-1]:
+                            os.rmdir(dirname)
+                    except Exception:
+                        pass
+
+                    module.fail_json(msg="Error mounting %s: %s" % (name, msg))
+            elif state == 'present':
+                name, changed = set_mount(module, args)
+            elif state == 'remounted':
+                if not module.check_mode:
+                    res, msg = remount(module, args)
+
+                    if res:
+                        module.fail_json(msg="Error remounting %s: %s" % (name, msg))
 
-        if (
-                ismount(name) or
-                is_bind_mounted(
-                    module, linux_mounts, name, args['src'], args['fstype'])):
-            if changed and not module.check_mode:
-                res, msg = remount(module, args)
                 changed = True
-        else:
-            changed = True
-
-            if not module.check_mode:
-                res, msg = mount(module, args)
-
-        if res:
-            # Not restoring fstab after a failed mount was reported as a bug,
-            # ansible/ansible#59183
-            # A non-working fstab entry may break the system at the reboot,
-            # so undo all the changes if possible.
-            try:
-                write_fstab(module, backup_lines, args['fstab'])
-            except Exception:
-                pass
-
-            try:
-                for dirname in dirs_created[::-1]:
-                    os.rmdir(dirname)
-            except Exception:
-                pass
-
-            module.fail_json(msg="Error mounting %s: %s" % (name, msg))
-    elif state == 'present':
-        name, changed = set_mount(module, args)
-    elif state == 'remounted':
-        if not module.check_mode:
-            res, msg = remount(module, args)
-
-            if res:
-                module.fail_json(msg="Error remounting %s: %s" % (name, msg))
+            else:
+                module.fail_json(msg='Unexpected position reached')
 
-        changed = True
-    else:
-        module.fail_json(msg='Unexpected position reached')
+    except LockTimeout:
```

Comment:
@quidame As I haven't tested the functionality wanted to check if LockTimeout raises an exception with a msg and if yes can we not use the respective exception msg along with the custom msg.

Reply:
hi @justjais, thanks for reviewing.
I did what you suggest, this gives:
```
fatal: [test1.jeu]: FAILED! => {"changed": false, "msg": "another instance of the module holds the fstab lock: timeout (2 sec)"}
```
(in fact it was `0.5 sec` because I had to decrease the `lock_timeout` in the code to this value to reach it. I assume that when the value is `2`, it displays `2 sec` or something very close)